components/movementsensor/adxl345/adxl345.go

```diff
@@ -76,7 +76,7 @@ type adxl345 struct {
 	// Lock the mutex when you want to read or write either the acceleration or the last error.
 	mu                 sync.Mutex
 	linearAcceleration r3.Vector
-	lastError          error
+	err                movementsensor.LastError
```

Comment:
[test-req] let's test with a `imu-wit`, `gps-nmea`, `accel-adxl345`, and `gyro-mpu6050`.

Reply:
:astonished: Excellent idea! I've got [this robot](https://app.viam.com/robot?id=ec50486f-92db-4364-b77e-59df960ad5ad&tab=control&part=df230c5c-0dcc-4d80-8a84-1e416e015603) currently hooked up with one of each of those. The Wit IMU periodically has spontaneous errors, but it recovers from them afterwards. The other 3 typically don't have errors, but if I unplug the I2C clock pin, they get errors, and then if I plug it back in, the errors disappear.

Compare this to running on the `main` branch: surprisingly, this has the same behavior! I absolutely didn't expect that, and currently have no explanation for how the errors disappear (even knowing that they _do_ disappear, I still can't see why). 

I kinda want to merge this branch because I think it's clearer what's going on, but I recognize that the old version appears to work as expected even if I don't understand why.